### PR TITLE
Fail test when calling function twice

### LINKED ISSUE #1195: Check if FairDetector::Initialize() is called twice
When running a transport simulation the Initialize() function of FairDetector is called twice. Once it is called from FairMCApplication, the second time it is called from TVirtualMC directly. The second call was added implicitly with the code changes and is the new default way to call Initialize() function of FairDetector.

Add a new test or change an existing test which crashes when the Initilize() function is called twice.

### PR DESCRIPTION
**Pull request by fuhlig1**
Change the Initialize function such that it asserts when the function is
called twice.

Closes #1195

Checklist:

* [ x] Rebased against `dev` branch
* [ x] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [ x] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

File: examples/simulation/Tutorial1/src/FairTutorialDet1.cxx [hunk 1]
```diff
@@ -68,7 +71,14 @@ FairTutorialDet1::~FairTutorialDet1()
...
+    LOG(info) << "Initializing FairTutorialDet1";
+
+    // Prevent duplicate initialisation
+    assert(!fIsInitialised);
+
     FairDetector::Initialize();
+    fIsInitialised = kTRUE;
     /*
   FairRuntimeDb* rtdb= FairRun::Instance()->GetRuntimeDb();
   FairTutorialDet1GeoPar* par=(FairTutorialDet1GeoPar*)(rtdb->getContainer("FairTutorialDet1GeoPar"));
```
Comment (ChristianTackeGSI) [inline examples/simulation/Tutorial1/src/FairTutorialDet1.cxx:84]: Maybe remove the commented code while you're changing this anyway?

**Review by ChristianTackeGSI — APPROVED** [review]
Some ideas. Can be merged as is, if you don't like those ideas.

File: examples/simulation/Tutorial1/src/FairTutorialDet1.h [hunk 2]
```diff
@@ -95,6 +95,8 @@ class FairTutorialDet1 : public FairDetector
     Double32_t fLength;    //!  length
     Double32_t fELoss;     //!  energy loss
 
+    Bool_t fIsInitialised;   //!  prevent double initilization
```
[the remark this answers is not in the document]
Comment (fuhlig1) [reply examples/simulation/Tutorial1/src/FairTutorialDet1.h:98]: @ChristianTackeGSI, 

the ideas are fine but I would prefer to move the PR as is.
I also thought about them but in my opinion this should be done in separate pull requests which move the code to more modern C++.

[the remark this answers is not in the document]
Comment (ChristianTackeGSI) [reply examples/simulation/Tutorial1/src/FairTutorialDet1.h:98]: Then go ahead and merge this one!

**Pull request merged**